src/lib/teads-aws/index.ts

```diff
@@ -0,0 +1,244 @@
+  }
+
+  /**
+   * Returns model identifier
+   */
+  modelIdentifier(): string {
+    return 'teads.cloud.sci';
+  }
+
+  /**
+   * Standardize the instance metrics for all the providers
+   *
+   * Maps the instance metrics to a standard format (min, max, idle, 10%, 50%, 100%) for all the providers
+   */
+  standardizeInstanceMetrics() {
+    AWS_INSTANCES.forEach((instance: KeyValuePair) => {
+      const cpus = parseInt(instance['Instance vCPU'], 10);
+      this.computeInstances[instance['Instance type']] = {
+        consumption: {
+          idle: parseFloat(instance['Instance @ Idle'].replace(',', '.')),
+          tenPercent: parseFloat(instance['Instance @ 10%'].replace(',', '.')),
+          fiftyPercent: parseFloat(
+            instance['Instance @ 50%'].replace(',', '.')
+          ),
+          hundredPercent: parseFloat(
+            instance['Instance @ 100%'].replace(',', '.')
+          ),
+        },
+        vCPUs: cpus,
+        maxvCPUs: parseInt(instance['Platform Total Number of vCPU'], 10),
+        name: instance['Instance type'],
+      } as KeyValuePair;
+    });
+    AWS_EMBODIED.forEach((instance: KeyValuePair) => {
+      this.computeInstances[instance['type']].embodiedEmission =
+        instance['total'];
+    });
+  }
+
+  /**
+   * Calculates the energy consumption for a single observation
+   * requires
+   *
+   * duration: duration of the observation in seconds
+   * cpu: cpu usage in percentage
+   * datetime: ISO 8601 datetime string
+   *
+   * Uses a spline method for AWS and linear interpolation for GCP and Azure
+   */
+  private calculateEnergy(observation: KeyValuePair) {
+    if (
+      !('duration' in observation) ||
+      !('cpu' in observation) ||
+      !('datetime' in observation)
+    ) {
+      throw new Error(
+        'Required Parameters duration,cpu,datetime not provided for observation'
+      );
+    }
+
+    //    duration is in seconds
+    const duration = observation['duration'];
+
+    //    convert cpu usage to percentage
+    const cpu = observation['cpu'] * 100.0;
+
+    //  get the wattage for the instance type
+
+    const x = [0, 10, 50, 100];
+
+    const y: number[] = [
+      this.computeInstances[this.instanceType].consumption.idle ?? 0,
+      this.computeInstances[this.instanceType].consumption.tenPercent ?? 0,
+      this.computeInstances[this.instanceType].consumption.fiftyPercent ?? 0,
+      this.computeInstances[this.instanceType].consumption.hundredPercent ?? 0,
+    ];
+
+    const spline = new Spline(x, y);
+
+    let wattage = 0.0;
+    if (this.interpolation === Interpolation.SPLINE) {
+      wattage = spline.at(cpu);
+    } else if (this.interpolation === Interpolation.LINEAR) {
+      let min = 0;
+      let max = 1;
+      if (cpu > 10 && cpu <= 50) {
+        min = 1;
+        max = 2;
+      }
+      if (cpu > 50 && cpu <= 100) {
+        min = 2;
+        max = 3;
+      }
+      wattage = y[0] + ((y[max] - y[min]) / (x[max] - x[min])) * cpu;
```

Comment:
Hmm are you sure it's not just y[min] + ..... ?

Reply:
@jawache I am implementing an improved mechanism. I will copy it over from Teads CPU.